hotrod/src/main/java/io/hyperfoil/hotrod/steps/HotRodRequestBuilder.java

```diff
@@ -0,0 +1,160 @@
+import io.hyperfoil.api.config.Name;
+import io.hyperfoil.api.config.Step;
+import io.hyperfoil.api.config.StepBuilder;
+import io.hyperfoil.api.session.Session;
+import io.hyperfoil.core.builders.BaseStepBuilder;
+import io.hyperfoil.core.generators.StringGeneratorBuilder;
+import io.hyperfoil.core.generators.StringGeneratorImplBuilder;
+import io.hyperfoil.core.metric.MetricSelector;
+import io.hyperfoil.core.metric.PathMetricSelector;
+import io.hyperfoil.core.metric.ProvidedMetricSelector;
+import io.hyperfoil.core.steps.StatisticsStep;
+import io.hyperfoil.function.SerializableFunction;
+import io.hyperfoil.hotrod.api.HotRodOperation;
+import io.hyperfoil.hotrod.resource.HotRodResource;
+
+/**
+ * Issues a HotRod request and registers handlers for the response.
+ */
+@MetaInfServices(StepBuilder.class)
+@Name("hotrodRequest")
+public class HotRodRequestBuilder extends BaseStepBuilder<HotRodRequestBuilder> {
+
+   private HotRodOperationBuilder operation;
+   private StringGeneratorBuilder cacheName;
+   private MetricSelector metricSelector;
+   private StringGeneratorBuilder key;
+   private StringGeneratorBuilder value;
+
+   @Override
+   public void prepareBuild() {
+      if (metricSelector == null) {
+         String sequenceName = Locator.current().sequence().name();
+         metricSelector = new ProvidedMetricSelector(sequenceName);
+      }
+   }
+
+   @Override
+   public List<Step> build() {
+      int stepId = StatisticsStep.nextId();
+      HotRodResource.Key key = new HotRodResource.Key();
+      SerializableFunction<Session, String> keyGenerator = this.key != null ? this.key.build() : null;
+      SerializableFunction<Session, String> valueGenerator = this.value != null ? this.value.build() : null;
+      HotRodRequestStep step = new HotRodRequestStep(stepId, key, operation.build(), cacheName.build(), metricSelector,
+            keyGenerator, valueGenerator);
+      HotRodResponseStep secondHotRodStep = new HotRodResponseStep(key);
+      return Arrays.asList(step, secondHotRodStep);
+   }
+
+   /**
+    * Requests statistics will use this metric name.
+    *
+    * @param name Metric name.
+    * @return Self.
+    */
+   public HotRodRequestBuilder metric(String name) {
+      return metric(new ProvidedMetricSelector(name));
+   }
+
+   public HotRodRequestBuilder metric(ProvidedMetricSelector selector) {
+      this.metricSelector = selector;
+      return this;
+   }
+
+   /**
+    * Allows categorizing request statistics into metrics based on the request path.
+    *
+    * @return Builder.
+    */
+   public PathMetricSelector metric() {
+      PathMetricSelector selector = new PathMetricSelector();
+      this.metricSelector = selector;
+      return selector;
+   }
+
+   public StringGeneratorImplBuilder<HotRodRequestBuilder> cacheName() {
+      StringGeneratorImplBuilder<HotRodRequestBuilder> builder = new StringGeneratorImplBuilder<>(this, false);
+      cacheName(builder);
+      return builder;
+   }
```

Comment:
Seeing the missing empty lines I realized that the checkstyle is not running in this module, please add
```xml
<plugin>
    <groupId>org.apache.maven.plugins</groupId>
    <artifactId>maven-checkstyle-plugin</artifactId>
</plugin>
```
to the pom.xml